core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -600,6 +600,12 @@ private[spark] class AppStatusListener(
     liveUpdate(task, now)
 
     Option(liveStages.get((event.stageId, event.stageAttemptId))).foreach { stage =>
+      if (event.taskInfo.speculative) {
+        stage.speculationStageSummary.numActiveTasks += 1
+        stage.speculationStageSummary.numTasks += 1
+      }
+      maybeUpdate(stage.speculationStageSummary, now)
```

Comment:
Should we update only if the task is speculative?

Reply:
Not necessary, update can happen even if the current task is not speculative and if the update period is configured to a different time. 

What we could do is change it to `update` and not `maybeUpdate` if the task is speculative but those may be too many updates. What do you think ?

Reply:
How about:
```suggestion
      if (stage.speculationStageSummary.numTasks > 0) {
        maybeUpdate(stage.speculationStageSummary, now)
      }
```
Otherwise, the metrics `speculationSummary` is always present even though a stage has no speculative tasks.
```
$ curl http://localhost:4040/api/v1/applications/local-1637337648156/stages/0/
...
  "speculationSummary" : {
    "numTasks" : 0,
    "numActiveTasks" : 0,
    "numCompletedTasks" : 0,
    "numFailedTasks" : 0,
    "numKilledTasks" : 0
  },
...
```

Reply:
ok, changed it.